cpp/src/arrow/util/hash-util.h

```diff
@@ -72,80 +72,80 @@ class HashUtil {
 #endif
 
 #ifdef ARROW_HAVE_ARMV8_CRYPTO
-  /* Arm64 Parallel crc32 computation
-   *
-   * Algorithm comes from Intel whitepaper:
-   * crc-iscsi-polynomial-crc32-instruction-paper
-   * Implementation porting from Ceph:crc32c_aarch64.c
-   */
+/* Crc32c Parallel computation
+ *   Algorithm comes from Intel whitepaper:
+ *   crc-iscsi-polynomial-crc32-instruction-paper
+ *
+ * Input data is divided into three equal-sized blocks
+ *   Three parallel blocks (crc0, crc1, crc2) for 1024 Bytes
+ *   One Block: 42(BLK_LENGTH) * 8(step length: crc32c_u64) bytes
+ */
+#define BLK_LENGTH 42
   static uint32_t Armv8CrcHashParallel(const void* data, int32_t nbytes, uint32_t crc) {
-    uint32_t crc0, crc1, crc2;
-    const poly64_t k1 = 0xe417f38a, k2 = 0x8f158014;
-    uint64_t t0, t1;
-    const uint8_t* buffer = reinterpret_cast<const uint8_t*>(data);
-
-    /* Process per block size of 1024 Bytes
-     * A block size = 8 + 42*3*sizeof(uint64_t) + 8 = 1024
-     */
-    while ((nbytes -= 1024) >= 0) {
-      /* Prefetch 3*1024 data for avoiding L2 cache miss */
-      PREF1KL2(buffer, 1024 * 3);
-
-      /* Do first 8 bytes here for better pipelining */
-      crc0 = __crc32cd(crc, *(const uint64_t*)buffer);
-      crc1 = 0;
-      crc2 = 0;
-      buffer += sizeof(uint64_t);
-
-      /* Process block inline
-       * Process crc0 last to avoid dependency with above
+    const uint8_t* buf8;
+    const uint64_t* buf64 = (const uint64_t*)data;
```

Comment:
Can you use C++-style casts in this file (`reinterpret_cast` and `static_cast`)?

Reply:
OK. Done.